subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -187,12 +207,30 @@ __net_socket struct tls_context {
 #endif /* CONFIG_MBEDTLS */
 };
 
+
 /* A global pool of TLS contexts. */
 static struct tls_context tls_contexts[CONFIG_NET_SOCKETS_TLS_MAX_CONTEXTS];
 
+static struct tls_session_cache client_cache[CONFIG_NET_SOCKETS_TLS_MAX_CLIENT_SESSION_COUNT];
+
+#if defined(MBEDTLS_SSL_CACHE_C)
+static mbedtls_ssl_cache_context server_cache;
+#endif
+
 /* A mutex for protecting TLS context allocation. */
 static struct k_mutex context_lock;
 
+static void tls_session_cache_reset(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(client_cache); i++) {
+		if (client_cache[i].session != NULL) {
+			mbedtls_free(client_cache[i].session);
+		}
+	}
+
+	(void)memset(client_cache, 0, sizeof(client_cache));
```

Comment:
shouldn't we add a compiler barrier here to avoit it be optimized out ?

Reply:
I'm not sure I follow, why would the compiler want to optimize this memset out?